main.go

```diff
@@ -496,12 +511,14 @@ func (s XcodeArchiveStep) xcodeArchive(opts xcodeArchiveOpts) (xcodeArchiveOutpu
 		archiveCmd.SetCustomBuildAction("clean")
 	}
 
-	xcconfigWriter := xcconfig.NewWriter(s.pathProvider, s.fileManager)
-	xcconfigPath, err := xcconfigWriter.Write(opts.XcconfigContent)
-	if err != nil {
-		return out, fmt.Errorf("failed to write xcconfig file contents: %w", err)
+	if opts.XcconfigContent != "" {
+		xcconfigWriter := xcconfig.NewWriter(s.pathProvider, s.fileManager, s.pathChecker)
+		xcconfigPath, err := xcconfigWriter.Write(opts.XcconfigContent)
```

Comment:
Just want to double check this part. This is the `Write` function from the config writer:
```
func (w writer) Write(input string) (string, error) {
	if w.isPath(input) {
		pathExists, err := w.pathChecker.IsPathExists(input)
		if err != nil {
			return "", fmt.Errorf(err.Error())
		}
		if !pathExists {
			return "", fmt.Errorf("provided xcconfig file path doesn't exist: %s", input)
		}
		return input, nil
	}

	dir, err := w.pathProvider.CreateTempDir("")
	if err != nil {
		return "", fmt.Errorf("unable to create temp dir for writing XCConfig: %v", err)
	}
	xcconfigPath := filepath.Join(dir, "temp.xcconfig")
	if err = w.fileManager.Write(xcconfigPath, input, 0644); err != nil {
		return "", fmt.Errorf("unable to write XCConfig content into file: %v", err)
	}
	return xcconfigPath, nil
}
```
If I read it correctly then if the user provided an xcconfig path as input then it returns relative and if not then it returns an absolute path. Can Xcode handle both types?

Reply:
Checked both are supported by Xcode and both can be provided as input.
We could convert the xconfig path to absolute when parsing the Step inputs to be consistent tough.